go/src/lib.rs

```diff
@@ -453,6 +453,20 @@ fn valkey_value_to_command_response(value: Value) -> RedisResult<CommandResponse
             command_response.response_type = ResponseType::Map;
             Ok(command_response)
         }
+        Value::Set(array) => {
```

Comment:
You can update line 418 to be `Value::Array(array) | Value::Set(array) => {` and remove duplicated code

Reply:
In my opinion, we should keep the `Sets` as a separate type in the `ResponseType` enum as it is being handled in a different way in GO to ensure we are expecting the correct type from valkey and have the correct type validations. Wdyt?

Reply:
I'm ok with that if we process set not the same way as array. Otherwise it could be a code duplicating.

Reply:
@umit Please add a `ResponseType` for `Sets` and validate it in the Go code as we are handling arrays and sets separately. 

```
#[repr(C)]
#[derive(Debug, Default)]
pub enum ResponseType {
    #[default]
    Null = 0,
    Int = 1,
    Float = 2,
    Bool = 3,
    String = 4,
    Array = 5,
    Map = 6,
    Sets = 7,
}
```

The values can still be sent in `array_value` and no change is required in the `CommandResponse`.

Reply:
@janhavigupta007 if you insist on keeping both cases separately, I would recommend that we move the `Array` code into a separate function and call it in both cases - until they diverge

Reply:
> I'm ok with that if we process set not the same way as array. Otherwise it could be a code duplicating.

@janhavigupta007 I am on the same page as @Yury-Fridlyand. We can keep it as is now.

Reply:
We have introduced the `ResponseType` enum in order to validate in GO code that the expected type is being sent from the Valkey engine compared to what is expected.

Since the handling of `Set` is different from `Array` in GO code unlike `BulkString, SimpleString, VerbatimString`, we need to validate that the engine is sending `Set` while processing `Set` response. Otherwise, we might process `Array` response as `Set` which won't be ideal and we might also end up losing data as processing for `Set` will eliminate the duplicates or throw unexpected exceptions.

Can we please propagate the `Set` responseType to GO?

Reply:
@janhavigupta007 I've added `Set` response type.